addon/components/horizontal-bar-chart.js

```diff
@@ -244,19 +246,26 @@ export default ChartComponent.extend(FloatingTooltipMixin,
       },
       y: this.get('barThickness') / 2,
       dy: '.35em',
-      'text-anchor': (d) => d.value < 0 ? 'end' : 'start',
+      'text-anchor': (d) => {
+        if ((d.value < 0) || ((d.value === 0) && this.get('hasNegativeValues') && !this.get('hasPositiveValues'))) {
```

Comment:
I'd add a helper function for this. Something like `_shouldLabelGoLeft(d)` which returns true if it's on the left and false otherwise.